transfers/pre-transfer/fix_unlv_checksums.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python
+
+from __future__ import print_function
+
+import sys
+import os
+
+
+HASHFUNC_EXT = 'md5'
+
+
+def find_objects(path, ignore_dirs=None):
+    for name in os.listdir(path):
+        full_path = os.path.join(path, name)
+        if full_path in (ignore_dirs,):
+            continue
+        if os.path.isdir(full_path):
+            for entry in find_objects(full_path, ignore_dirs):
+                yield entry
+        elif os.path.isfile(full_path):
+                yield full_path
+
+
+def extract_checksum(filename, metadata_path):
+    filename = '{}.{}'.format(os.path.join(metadata_path, filename),
+                              HASHFUNC_EXT)
+    checksum = None
+    try:
+        with open(os.path.join(metadata_path, filename), 'r') as f:
+            checksum = f.readline().strip()
+    except:
+        pass
+    return checksum
+
+
+def main(transfer_path):
+    """
+    Generate the standard checksum file in a transfer where every object has
+    its checksum stored in a separate text file under the metadata directory,
+    e.g. transfer-A is the original transfer and transfer-B is the resulting
+    transfer after running this script.
+
+       .
+       ├── transfer-A
+       │   ├── foobar.mp3
+       │   └── metadata
+       │       └── foobar.mp3.md5
+       └── transfer-B
+        ├── foobar.mp3
+        └── metadata
+            ├── checksum.md5
+            └── foobar.mp3.md5
+
+    """
+    transfer_path = os.path.abspath(transfer_path)
+    metadata_path = os.path.join(transfer_path, 'metadata')
+
+    if not os.path.exists(metadata_path):
+        print('metadata directory cannot be found: {}'.format(metadata_path),
+              file=sys.stderr)
+        return 1
+
+    checksum_file = os.path.join(metadata_path, 'checksum.{}'.format(
+                                 HASHFUNC_EXT))
+    with open(checksum_file, 'w') as f:
+        for item in find_objects(transfer_path, ignore_dirs=metadata_path):
+            basename = os.path.basename(item)
+            hashsum = extract_checksum(basename, metadata_path)
+            if hashsum is None:
+                print('Cannot find checksum of {}'.format(basename),
+                      file=sys.stderr)
```

Comment:
`continue` after this?

Print knows how to convert things to strings, so calling .format on them isn't needed.  This could be `print('Cannot find checksum of', basename, file=sys.stderr)`